jerry-core/ecma/base/ecma-helpers-value.c

```diff
@@ -136,9 +160,8 @@ ecma_is_value_null (ecma_value_t value) /**< ecma value */
 inline bool __attr_pure___ __attr_always_inline___
 ecma_is_value_boolean (ecma_value_t value) /**< ecma value */
 {
-  return (ecma_get_value_type_field (value) == ECMA_TYPE_SIMPLE
-          && (ecma_get_value_value_field (value) == ECMA_SIMPLE_VALUE_TRUE
-              || ecma_get_value_value_field (value) == ECMA_SIMPLE_VALUE_FALSE));
+  return (ecma_is_value_equal_to_simple_value (value, ECMA_SIMPLE_VALUE_TRUE)
+          || ecma_is_value_equal_to_simple_value (value, ECMA_SIMPLE_VALUE_FALSE));
```

Comment:
There are some plans for a follow-up work here. But I try to simplify this a bit.